python/semantic_kernel/connectors/ai/mistral_ai/services/mistral_ai_text_embedding.py

```diff
@@ -0,0 +1,88 @@
+from mistralai.models.embeddings import EmbeddingResponse
+from numpy import array, ndarray
+from pydantic import ValidationError
+
+from semantic_kernel.connectors.ai.embeddings.embedding_generator_base import EmbeddingGeneratorBase
+from semantic_kernel.connectors.ai.mistral_ai.settings.mistral_ai_settings import MistralAISettings
+from semantic_kernel.exceptions.service_exceptions import ServiceInitializationError, ServiceResponseException
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+
+@experimental_class
+class MistralAITextEmbedding(EmbeddingGeneratorBase):
+    """Mistral AI Inference Text Embedding Service."""
+
+    client: MistralAsyncClient
+
+    def __init__(
+        self,
+        ai_model_id: str | None = None,
+        api_key: str | None = None,
+        service_id: str | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = None,
+        client: MistralAsyncClient | None = None,
+    ) -> None:
+        """Initialize the Mistral AI Text Embedding service.
+
+        If no arguments are provided, the service will attempt to load the settings from the environment.
+        The following environment variables are used:
+        - MISTRALAI_API_KEY
+        - MISTRALAI_EMBEDDING_MODEL_ID
+
+        Args:
+            ai_model_id: (str | None): A string that is used to identify the model such as the model name. 
+            api_key (str | None): The API key for the Mistral AI service deployment. 
+            service_id (str | None): Service ID for the embedding completion service. 
+            env_file_path (str | None): The path to the environment file. 
+            env_file_encoding (str | None): The encoding of the environment file. 
+            client (MistralAsyncClient | None): The Mistral AI client to use. 
+
+        Raises:
+            ServiceInitializationError: If an error occurs during initialization.
+        """
+        try:
+            mistralai_settings = MistralAISettings.create(
+                api_key=api_key,
+                embedding_model_id=ai_model_id,
+                env_file_path=env_file_path,
+                env_file_encoding=env_file_encoding,
+            )
+        except ValidationError as e:
+            raise ServiceInitializationError(f"Failed to validate Mistral AI settings: {e}") from e
+
+        if not mistralai_settings.embedding_model_id:
+            raise ServiceInitializationError("The MistralAI embedding model ID is required.")
+
+        if not client:
+            client = MistralAsyncClient(
+                api_key=mistralai_settings.api_key.get_secret_value()
+            )
+
+        super().__init__(
+            service_id=service_id or mistralai_settings.embedding_model_id,
+            ai_model_id=ai_model_id or mistralai_settings.embedding_model_id,
+            client=client,
+        )
+
+    async def generate_embeddings(self, texts: list[str], **kwargs: Any) -> ndarray:
```

Comment:
We just made some changes to the embeddings, preferably we support that here as well, includes using settings and adding a raw embeddings method that doesnt cast to a ndarray. Tao also found out that you don't have to cast the inner list seperately, so that additional loop isn't needed

Reply:
I divided the embeddings into 2 Methods `generate_raw_embeddings` and `generate_embeddings` , i can skip the second cast for sure, but i am not sure if i can really skip the for Loop here

This is the Structure of the Mistral Object.
```

class EmbeddingObject(BaseModel):
    object: str
    embedding: List[float]
    index: int


class EmbeddingResponse(BaseModel):
    id: str
    object: str
    data: List[EmbeddingObject]
    model: str
    usage: UsageInfo


```
I think i need the iteration to get the Embedding Object out of the Embedding Response. I tried with some ideas to remove the for loop, but couldn't really figure out if i can eliminate the loop here.

Reply:
@eavanvalkenburg, can you have another look, please?